presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java

```diff
@@ -46,17 +48,18 @@
     private final Specification specification;
     private final int preSortedOrderPrefix;
     private final Map<Symbol, Function> windowFunctions;
+    private final Map<Function, Frame> frames;
     private final Optional<Symbol> hashSymbol;
 
-    @JsonCreator
     public WindowNode(
-            @JsonProperty("id") PlanNodeId id,
-            @JsonProperty("source") PlanNode source,
-            @JsonProperty("specification") Specification specification,
-            @JsonProperty("windowFunctions") Map<Symbol, Function> windowFunctions,
-            @JsonProperty("hashSymbol") Optional<Symbol> hashSymbol,
-            @JsonProperty("prePartitionedInputs") Set<Symbol> prePartitionedInputs,
-            @JsonProperty("preSortedOrderPrefix") int preSortedOrderPrefix)
+            PlanNodeId id,
+            PlanNode source,
+            Specification specification,
+            Map<Symbol, Function> windowFunctions,
+            Map<Function, Frame> frames,
```

Comment:
I'm still running some tests locally to make sure that my changes are valid, but since @maciejgrzybek has already pointed you over there, feel free to take a gander.